include/glow/ExecutionEngine/ExecutionEngine.h

```diff
@@ -118,8 +118,8 @@ class ExecutionEngine final {
   /// Make \p networkName the function name for
   /// the entry point of the network and prepend all generated
   /// files with this name.
-  void save(Function *F, const CompilationOptions &opts,
-            llvm::StringRef outputDir, llvm::StringRef networkName);
+  void save(Function *&F, CompilationOptions &opts, llvm::StringRef outputDir,
```

Comment:
Why did you remove the const? Are we modifying the compilation options inside? This will make it really difficult to track things around.

Reply:
The only thing that's being modified is the `LoweredInfoMap` -- I could move that to be a pointer and keep `CompilationOptions` `const`. Then the `LoweredInfoMap` would need to be explicitly kept in the caller's scope or higher.